mitodl/mit-open
# Add profile edit page

### What are the relevant tickets?




Closes https://github.com/mitodl/hq/issues/4565
Fixes https://github.com/mitodl/hq/issues/4504

### Description (What does it do?)

This adds the profile edit page under the dashboard.

- It significantly refactors the field implementations for onboarding so that they can be reused on this page.
- It moves `ChoiceBox` into `ol-components` and makes it more flexible and configurable so that it can be used on both pages.
- I also made the onboarding steps optional since that bug had been reported and I was in that code changing things and the profile edit page needed to behave the same.

### Screenshots (if appropriate):

![Screenshot 2024-06-14 at 16-29-42 User Home](https://github.com/mitodl/mit-open/assets/28598/53c8bcee-c751-4a63-b4dd-6898825fd789)
![Screenshot 2024-06-14 at 16-30-27 User Home](https://github.com/mitodl/mit-open/assets/28598/8b6df40b-dc18-41fc-b5e9-4902836d2a75)
![Screenshot 2024-06-14 at 16-30-15 User Home](https://github.com/mitodl/mit-open/assets/28598/fc8ab338-d199-4103-9346-ea92c5853fe9)


### How can this be tested?

- You should be able to navigate to the profile edit page under the dashboard and edit your profile there.
- You should still be able to successfully complete the onboarding flow (`/onboarding`).
